x-pack/plugins/lens/public/xy_visualization_plugin/xy_suggestions.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { partition } from 'lodash';
+import { Position } from '@elastic/charts';
+import { XYArgs } from './xy_expression';
+import { SuggestionRequest, VisualizationSuggestion, TableColumn, TableSuggestion } from '../types';
+
+/**
+ * Generate suggestions for the xy chart.
+ *
+ * @param opts
+ */
+export function getSuggestions(
+  opts: SuggestionRequest<XYArgs>
+): Array<VisualizationSuggestion<XYArgs>> {
+  return Object.entries(opts.tables)
+    .filter(
+      ([_id, { isMultiRow, columns }]) =>
+        // We only render line charts for multi-row queries.
+        isMultiRow &&
+        // We require at least two columns: one for x and at least one for y
+        columns.length > 1 &&
+        // y columns must be numeric
+        columns.find(col => col.operation.dataType === 'number')
```

Comment:
The comment and implementation here are still not quite in sync. I suggested using `.some` instead of `.find` because all you care about is whether there exists a numeric column- this doesn't check whether it's on the Y axis at all, it just validates that there is a number column.

I would suggest removing all the comments:

```

      ([_id, { isMultiRow, columns }]) =>
        isMultiRow &&
        columns.length > 1 &&
        columns.some(col => col.operation.dataType === 'number')
```

Reply:
I added the comments in response to one of your previous comments on this. I think it makes sense to have a single comment on why we're filtering the way we are. Agree about `some` note sure why I used `find`.